core/src/main/java/org/jruby/RubyTime.java

```diff
@@ -905,21 +897,20 @@ public long microseconds() {
     	return getMicros();
     }
 
-
     /**
      * Return the nano-seconds of this time.
      * @return nano seconds (only)
      */
-    public long getNanos() {
-        return (getTimeInMillis() % 1000) * 1_000_000 + getNSec();
+    public int getNanos() {
```

Comment:
Why these signature changes? They worry me like the RubyString.length change but in this case there's no alternative you can call with the old signature.